### PR TITLE
Update PHP CS Fixer

### PR DESCRIPTION
**Pull request by MathiasReker**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop 
| Description?  | /
| Type?         | improvement
| Category?     | CO
| BC breaks?    | Does it break backward compatibility? no
| Deprecations? | Does it deprecate an existing feature? no
| Fixed ticket? | /
| How to test?  | /

Rule | PR
-- | --
align_multiline_comment' =>   ['comment_type' => 'phpdocs_only'] | https://github.com/PrestaShop/PrestaShop/pull/11694
'array_indentation' => true | https://github.com/PrestaShop/PrestaShop/pull/11695
'combine_consecutive_issets' => true | https://github.com/PrestaShop/PrestaShop/pull/11696
'combine_consecutive_unsets' => true | https://github.com/PrestaShop/PrestaShop/pull/11697
'hash_to_slash_comment' => true | https://github.com/PrestaShop/PrestaShop/pull/11699
'logical_operators' => true | https://github.com/PrestaShop/PrestaShop/pull/11701
'no_short_echo_tag' => true | https://github.com/PrestaShop/PrestaShop/pull/11703
'no_alias_functions' => true | https://github.com/PrestaShop/PrestaShop/pull/11707
'modernize_types_casting' => true | https://github.com/PrestaShop/PrestaShop/pull/11708
'no_mixed_echo_print' => true | https://github.com/PrestaShop/PrestaShop/pull/11709
'no_closing_tag' => true | https://github.com/PrestaShop/PrestaShop/pull/11738
'single_blank_line_at_eof' => true | https://github.com/PrestaShop/PrestaShop/pull/11738

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/11718)

### CODE REVIEW

**Comment by MathiasReker** [comment]
WDYT?

**Comment by PierreRambaud** [comment]
@PrestaShop/prestashop-core-developers Need everyone feedback :)

**Comment by Quetzacoalt91** [comment]
I won't say no to additional checks, especially if they can be fixed by CS fixer. :)

**Comment by jolelievre** [comment]
This new setting seems good to me, and thank you for all the other PRs where you cleaned all the code!
I think we should wait to merge all the other PRs before this one.

**Comment by Quetzacoalt91** [comment]
Of course, this one should not pass the linter jobs until the other PRs are merged. :)

**Comment by AJenbo** [comment]
Have you guys considered setting up StyleCI? It runs php-cs-fixer as a service.

**Comment by Quetzacoalt91** [comment]
We are looking for tools, but haven't chosen yet. @eternoendless on its side heard about prettyci.com.

**Comment by AJenbo** [comment]
Hm I think prettyci.com might actually be a better choice especially if they respect the .php_cs rather then require a separate format.

**Comment by MathiasReker** [comment]
Please merge this PR.

**Comment by Quetzacoalt91** [comment]
Hi @MathiasReker,

In order to go on this PR and approve your changes, I restarted the Travis build. Unfortunately it appears the linter jobs are currently failing.
This means more changes must be brought on this PR to make it pass.

File: .php_cs.dist [hunk 1]
```diff
@@ -32,6 +32,16 @@ return PhpCsFixer\Config::create()
         'psr4' => false,
         'self_accessor' => false,
         'yoda_style' => null,
+        'align_multiline_comment' => [
+            'comment_type' => 'phpdocs_only',
+        ],
+        'array_indentation' => true, 
+        'combine_consecutive_issets' => true,
+        'combine_consecutive_unsets' => true,
+        'hash_to_slash_comment' => true,
```
Comment (eternoendless) [inline .php_cs.dist]: This has been deprecated, we should use `single_line_comment_style`

**Review by eternoendless — COMMENTED** [review]
Looks like some rules are missing:

- no_short_echo_tag
- no_alias_functions
- modernize_types_casting
- no_mixed_echo_print

**Comment by MathiasReker** [comment]
To migrate to all those rules, I just commited, run CS fixer with those settings:

`<?php

return PhpCsFixer\Config::create()
->setRiskyAllowed(true)
    ->setRules([
        'silenced_deprecation_error' => true,
        'error_suppression' => true,
        'fopen_flags' => true,
        'fopen_flag_order' => true,
        'implode_call' => true,
        'logical_operators' => true,
        'modernize_types_casting' => true,
        'no_alias_functions' => true,
        'no_homoglyph_names' => true,
        'non_printable_character' => true,
        'set_type_to_cast' => true,
        'string_line_ending' => true,
        'align_multiline_comment' => [
            'comment_type' => 'phpdocs_only',
        ],
        'array_indentation' => true,
        'binary_operator_spaces' => true,
        'combine_consecutive_issets' => true,
        'combine_consecutive_unsets' => true,
        'elseif' => true,
        'heredoc_to_nowdoc' => true,
        'method_chaining_indentation' => true,
        'multiline_comment_opening_closing' => true,
        'multiline_whitespace_before_semicolons' => [
            'strategy' => 'no_multi_line',
        ],
        'no_alternative_syntax' => true,
        'no_binary_string' => true,
        'no_short_echo_tag' => true,
            ])
    ->setLineEnding("\n");
`

**Comment by MathiasReker** [comment]
Can you please have a look at those PR's:

#11990
#11988
#11984
#11980
#11978
#11977
#11976
#11975
#11974
#11973
#11972
#11970
#11969
#11968
#11967
#11868
#11859
#11822
#11821
#11820
#11813
#11730
#10645

File: .php_cs.dist (hunk 1 above)
Comment (MathiasReker) [reply .php_cs.dist]: Solved.

**Comment by MathiasReker** [comment]
If some rules are missing now, they are missing because they are already a symfony rule :-)

File: .php_cs.dist [hunk 2]
```diff
@@ -33,7 +31,36 @@ return PhpCsFixer\Config::create()
         'psr4' => false,
         'self_accessor' => false,
         'yoda_style' => null,
+        'silenced_deprecation_error' => true,
+        'fopen_flags' => true,
```
Comment (PierreRambaud) [inline .php_cs.dist]: Already in Symfony risky

File: .php_cs.dist [hunk 3]
```diff
@@ -33,7 +31,36 @@ return PhpCsFixer\Config::create()
...
+        'no_alias_functions' => true,
+        'no_homoglyph_names' => true,
         'non_printable_character' => true,
+        'set_type_to_cast' => true,
+        'string_line_ending' => true,
+        'align_multiline_comment' => [
+            'comment_type' => 'phpdocs_only',
+        ],
+        'array_indentation' => true,
+        'binary_operator_spaces' => true,
```
Comment (PierreRambaud) [inline .php_cs.dist]: Already in Sf.

**Comment by PierreRambaud** [comment]
Look like you include all @Symfony:risky elements, as it says, it is risky :/ but use `'@Symfony:risky' => true` instead. And do not duplicate rules already in `@Symfony`, or `@Symfony:risky`

**Comment by MathiasReker** [comment]
@PierreRambaud I am not sure I included all risky symfony rules, I picked them one by one. None of included risky rules seems to have any negative affect on on this project. Are you sure would use `@Symfony:risky` ? What if Cs fixer is updated with new risky rules?

**Comment by PierreRambaud** [comment]
We have composer.lock which target a specific php-cs-fixer version. It is exactly the same as phpunit, Symfony and all dependencies. We control what we have, and so we should watch diff before updating dependencies.

**Comment by MathiasReker** [comment]
@PierreRambaud I updated the configuration. Wdyt?

```
        'native_constant_invocation' => false,
        'native_function_invocation' => false,
```

**Comment by MathiasReker** [comment]
WDYT?

I am not sure how to solve Travis CI lint.

**Comment by eternoendless** [comment]
New commits add more checks than were initially proposed. Can we roll them back and go with the initial suggestion? We can discuss adding others in another PR.

**Comment by MathiasReker** [comment]
@eternoendless I will do so. I am busy ATM. I will do it ASAP.

**Comment by MathiasReker** [comment]
I messed something up, trying to squash my commits. New PR: https://github.com/PrestaShop/PrestaShop/pull/12922

**Pull request closed without merging**